### PR TITLE
GS-375 Port media element over to Fozzie

### PR DESCRIPTION
**Pull request by 80sgreek**
Direct port of media element with mixin updates where necessary for compatibility.

### CODE REVIEW

File: src/scss/components/_media-element.scss [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ * Media element
+ * ===================================
+ * Example – SERP – Restaurant logos
+ *
+ * A common pattern that sees an image followed by a block of information.
+ *
+ */
+
+$mediaElement-img-width: 50px !default;
+$mediaElement-img-height: 50px !default;
```
Comment (kevinrodrigues) [inline src/scss/components/_media-element.scss]: Do you need this variable and the other height variants? They don't seem to be used anywhere here.

File: src/scss/components/_media-element.scss [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                display: none;
+            }
+        }
+    }
+
+    .mediaElement-text {
+        margin-top: 0;
+        @include font-size(small, false);
+    }
+    .mediaElement-text--large {
```
Comment (kevinrodrigues) [inline src/scss/components/_media-element.scss]: More of a question, should there be a line break before this selector?

**Comment by 80sgreek** [comment]
Just realised I need to update the format to include a 'c-' at the beginning of each class.

File: src/scss/components/_media-element.scss (hunk 1 above)
Comment (80sgreek) [reply src/scss/components/_media-element.scss]: @kevinrodrigues good point :)

I'll remove these, maybe they were added as an indicator of 'square logos'

**Comment by 80sgreek** [comment]
@kevinrodrigues I've also removed the !default flag from the width variables. They can be added back in the future when/if needed.

File: src/scss/components/_media-element.scss [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+ * Media element
+ * ===================================
+ * Example – SERP – Restaurant logos
+ *
+ * A common pattern that sees an image followed by a block of information.
+ *
+ */
+
+$mediaElement-img-width: 50px;
+
```
Comment (ashleynolan) [inline src/scss/components/_media-element.scss]: Super nit picky – any need for the spaces between these vars?

File: src/scss/components/_media-element.scss [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+$mediaElement-img-width--mid: 78px;
+
+$mediaElement-img-width--wide: 67px;
+
+
+// mediaElement module
+// Usual of having an image to the left and content to the right
```
Comment (ashleynolan) [inline src/scss/components/_media-element.scss]: This line doesn't read quite right.  Maybe should be `Usually consists of having…`

**Comment by ashleynolan** [comment]
@tommaton @80sgreek The way we decided was least confusing (as objects and components are so similar) was if objects only apply to native HTML tags.  So objects will be pretty minimal – things like `o-btn`, `o-input` that kind of thing.  Everything else should be a component.

**Comment by ashleynolan** [comment]
Looks good – just a couple of tiny documentation comments but otherwise good to merge

File: src/scss/components/_media-element.scss (hunk 3 above)
Comment (80sgreek) [reply src/scss/components/_media-element.scss]: It's a direct copy from INT CW, I can change it if you want but can we still aim to get this in today?

File: src/scss/components/_media-element.scss (hunk 4 above)
Comment (80sgreek) [reply src/scss/components/_media-element.scss]: It's a direct copy from INT CW, I can change it if you want but can we still aim to get this in today?

Comment (80sgreek) [reply src/scss/components/_media-element.scss]: Actually, I completely agree even if it is a cut and paste :laughing:

**Comment by 80sgreek** [comment]
@tommaton My understanding of the 'o-' / 'c-' was that it was that 'o-' was for classes on native tags like '<button />' or '<a />'. If I'm wrong someone should probably explain it to me again as I'm adding it for other people's benefit, not mine :smile:

**Comment by DamianMullins** [comment]
@tommaton @80sgreek I'll PM you the link to the CSS naming scheme (it's an internal link)

**Pull request merged**